docs/docker-compose.yaml

```diff
@@ -0,0 +1,40 @@
+# MODE="LIVE|QA|BUILD" PACKAGE=element_array_ephys UPSTREAM_REPO=https://github.com/datajoint/element-array-ephys.git HOST_UID=$(id -u) docker compose -f docs/docker-compose.yaml up --build
+#
+# navigate to http://localhost/
+version: "2.4"
+services:
+  docs:
+    build:
+      dockerfile: docs/.docker/Dockerfile
+      context: ../
+      args:
+        - PACKAGE
+    image: ${PACKAGE}-docs
+    environment:
+      - PACKAGE
+      - UPSTREAM_REPO
+      - MODE
+    volumes:
+      - ..:/main
+    user: ${HOST_UID}:anaconda
+    ports:
+      - 80:80
+    command:
+      - sh
+      - -c
+      - |
+        set -e
+        if echo "$${MODE}" | grep -i live &>/dev/null; then
+            mkdocs serve --config-file ./docs/mkdocs.yaml -a 0.0.0.0:80
+        elif echo "$${MODE}" | grep -iE "qa|build" &>/dev/null; then
+            git branch -D gh-pages || true
+            git fetch $${UPSTREAM_REPO} gh-pages:gh-pages || true
+            mike deploy --config-file ./docs/mkdocs.yaml -u $$(grep -oE '\d+\.\d+' /main/$${PACKAGE}/version.py) latest
+            mike set-default --config-file ./docs/mkdocs.yaml latest
+            if echo "$${MODE}" | grep -i qa &>/dev/null; then
+                mike serve --config-file ./docs/mkdocs.yaml -a 0.0.0.0:80
+            fi
```

Comment:
Summarizing for myself - lmk if I have it right
1. `live` is for active editing, dev
2. `qa` or `build` are  for deleting and rebuilding the upstream `gh-pages` branch, including setting `mike` versioning dropdown

Reply:
I'd make the following correction:

1. `live` is for active editing, dev. No versioning will be displayed here but your environment will `hot-reload` on saves which is convenient for previewing as you go.
2. `qa` is a final inspection check to see how it will all look with `mike` versioning. This is still only local and nothing is committed remotely with this step. `hot-reload` is not supported though which means changes will require `docker compose` down/up
3. `build` is necessary only for GHA. This step is only used right before deploying the actual website i.e. when you `push` a new `X.X.X` to make a release.